typespec-tests/src/main/java/com/cadl/specialheaders/EtagHeadersOptionalBodyAsyncClient.java

```diff
@@ -55,8 +55,6 @@ public final class EtagHeadersOptionalBodyAsyncClient {
      * <table border="1">
      * <caption>Header Parameters</caption>
      * <tr><th>Name</th><th>Type</th><th>Required</th><th>Description</th></tr>
-     * <tr><td>Content-Type</td><td>String</td><td>No</td><td>The content type. Allowed values:
-     * "application/json".</td></tr>
```

Comment:
This is deleted because TCGC now will add `content-type` as `application/json` by default. 
We previously always added content type if request body is optional, to avoid not adding `content-type` header when fire the call.
I think this is acceptable because it does not affect the runtime functionality. @weidongxu-microsoft  @srnagar what do you think?

Reply:
And the Impl now also sets the `content-type` header explicitly, right?

Here I think the case is that the request body is optional, and hence the Javadoc here reminds user to set the header in `requestOption` when they set the request body there.

Reply:
Yes, `impl` explicity set the `content-type` header: https://github.com/Azure/autorest.java/blob/7a5479dd75c522a9b02dad7d4ab072d112548b82/typespec-tests/src/main/java/com/cadl/specialheaders/implementation/EtagHeadersOptionalBodiesImpl.java#L150

Reply:
remove the content-type added by TCGC in emitter for optional body case.

Reply:
Fixed in 9e5146e7e931973a8c1c1d78a9b7bf4c909db772, and the generated file does not have diff now.